src/Data/Audience.elm

```diff
@@ -35,6 +42,129 @@ type alias Audience =
     }
 
 
+{-| Information of an `AudienceType` variant
+-}
+type alias AudienceTypeInfo msg =
+    { index : Int
+    , text : String
+    , icon : Html.Html msg
+    }
+
+
+
+-- Decoders
+
+
+{-| `Result` of the JSON parsing.
+It returns `Ok ( List Audience )` if the decoding was successful.
+It returns `Err ( ErrorCode, ErrorDesc )` if the decoding failed.
+-}
+audiences : Result ( String, String ) (List Audience)
+audiences =
+    Result.mapError
+        (\decoderError -> ( "JSON-A", "Parse audiencesJSON says: " ++ Json.Decode.errorToString decoderError ))
+        (Json.Decode.decodeString audiencesDecoder audiencesJSON)
+
+
+audiencesDecoder : Json.Decode.Decoder (List Audience)
+audiencesDecoder =
+    Json.Decode.field "data" (Json.Decode.list audienceDecoder)
+
+
+audienceDecoder : Json.Decode.Decoder Audience
+audienceDecoder =
+    Json.Decode.map4 Audience
```

Comment:
Using `Decode.mapN` is OK, but in my eyes the `mapN` is limited on large JSON objects (I think `map8` is the largest one). I'd prefer `Json.Decode.Pipeline` instead.

Reply:
@kraklin I believe I did use `map4` in my implementation of the challenge too - because for this small example it's enough, and the type won't grow (unlike in real apps :upside_down_face: ).